packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -815,6 +817,26 @@ export class StripeHelper {
     return this.fetchAllPlans();
   }
 
+  /**
+   * Verify that a given source country matches the plan currency.
+   */
+
+  isCurrencyCompatibleWithCountry(plan: AbbrevPlan, country: string): bool {
+    // TODO - Do these lists need to be mutually exclusive?
+    const currencyCountryMap = new Map([
+      [
+        'USD',
+        ['US', 'GB', 'NZ', 'MY', 'SG', 'CA', 'AS', 'GU', 'MP', 'PR', 'VI'],
+      ],
+      ['EUR', ['FR', 'DE']],
```

Comment:
This list needs to be a configuration variable in `config.ts` so that we can change it during deploys.